backtrace-library/src/main/java/backtraceio/library/common/ApplicationHelper.java

```diff
@@ -0,0 +1,74 @@
+package backtraceio.library.common;
+
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+
+import backtraceio.library.logger.BacktraceLogger;
+
+public class ApplicationHelper {
+    private static final transient String LOG_TAG = ApplicationHelper.class.getSimpleName();
+    /**
+     * Cached application name
+     */
+    private static String applicationName;
+
+    /**
+     * Cached application version
+     */
+    private static String applicationVersion;
+
+    /**
+     * Cached package name
+     */
+    private static String packageName;
+
+    /**
+     * Retrieves application name from context. The name will be cached over checks
+     * @param context application context
+     * @return application name
+     */
+    public static String getApplicationName(Context context) {
```

Comment:
All of your static methods requires Context so why not pass it through constructor?

Reply:
my thinking around this:

- initializing cache values once

there is a problem with this - you need to be sure its initialized at least once. You don't know at what point its initialized. Since we have a lot of components that can work together or separated, I have a small problem with this. The best moment is to initialize it during the client initialization, however what if you want to use only database options? 

Cache is necessary to avoid reflection costs.

Reply:
Moved it to the singleton so we should have optimal solution